wire-ios-data-model/Source/Model/User/ZMUser.m

```diff
@@ -679,7 +679,10 @@ + (ZMUser *)createSessionIfNeededInContext:(NSManagedObjectContext *)moc withSel
 
 + (ZMUser *)unboxSelfUserFromContextUserInfo:(NSManagedObjectContext *)moc
 {
-    ZMBoxedSelfUser *boxed = moc.userInfo[SelfUserKey];
+    __block ZMBoxedSelfUser *boxed;
+    [moc performBlockAndWait:^{
+        boxed = moc.userInfo[SelfUserKey];
+    }];
```

Comment:
Could this have unintended side effects? 🤷‍♂️ `userInfo` is an attribute of the context and crashes when accessed from the wrong thread.

Reply:
The original code assumes we're calling the method from the correct thread. I think that's why you may see that `ZMUser.selfUser(in:)` crashes if called outside of the context's queue. I don't see anything wrong with your changes.

Reply:
I think that's to be preferred